src/snowflake/cli/_plugins/workspace/commands.py

```diff
@@ -14,23 +14,73 @@
 
 from __future__ import annotations
 
+import logging
+
 import typer
+import yaml
 from snowflake.cli._plugins.nativeapp.artifacts import BundleMap
+from snowflake.cli._plugins.snowpark.commands import migrate_v1_snowpark_to_v2
+from snowflake.cli._plugins.streamlit.commands import migrate_v1_streamlit_to_v2
 from snowflake.cli._plugins.workspace.manager import WorkspaceManager
 from snowflake.cli.api.cli_global_context import get_cli_context
 from snowflake.cli.api.commands.decorators import with_project_definition
 from snowflake.cli.api.commands.snow_typer import SnowTyper
 from snowflake.cli.api.entities.common import EntityActions
 from snowflake.cli.api.output.types import MessageResult
+from snowflake.cli.api.project.definition_manager import DefinitionManager
+from snowflake.cli.api.secure_path import SecurePath
 
 ws = SnowTyper(
     name="ws",
-    hidden=True,
     help="Deploy and interact with snowflake.yml-based entities.",
 )
+log = logging.getLogger(__name__)
+
+
+@ws.command()
+def migrate(
+    accept_templates: bool = typer.Option(
+        False, "-t", "--accept-templates", help="Allows the migration of templates."
+    ),
+    **options,
+):
+    """Migrates the Snowpark and Streamlit project definition files form V1 to V2."""
+    pd = DefinitionManager().unrendered_project_definition
+
+    if pd.meets_version_requirement("2"):
+        return MessageResult("Project definition is already at version 2.")
+
+    if "<% ctx." in str(pd):
+        if not accept_templates:
+            raise ValueError(
```

Comment:
```suggestion
            raise ClickException(
```
This will make the result prettier